spaceMissionManager-persistance/src/main/java/cz/muni/fi/Entity/Spacecraft.java

```diff
@@ -17,42 +18,55 @@
 	private String name;
 
 	@OneToMany(fetch = FetchType.LAZY)
-	private List<CraftComponent> allComponents;
+	private List<CraftComponent> components;
 
 	@ManyToOne(fetch = FetchType.LAZY)
 	private Mission mission;
 
 
-
 	//METHODS
+	public Long getId() {
+		return id;
+	}
 
-	//TODO managing components - add, remove...
+	public void setId(Long id) {
+		this.id = id;
+	}
 
 	public String getName() {
 		return name;
 	}
 
-	public Long getId() {
-		return id;
-	}
-
 	public void setName(String name) {
 		this.name = name;
 	}
 
-	public List<CraftComponent> getAllComponents() {
-		return allComponents;
+	public void addComponent(CraftComponent c){
+		components.add(c);
```

Comment:
Here you are missing.
c.setSpacecraft(this);  @chlupaccom should add setSpacecraft method to CraftComponent entty
It is necessary to have it, because it connects Spacecraft with CraftComponents

Reply:
fix this and we can rebase

Reply:
does it need to be handle on this layer? Isn't this obligation of some higher lever in the architecture? But I guess you are right...so I fixed

Reply:
In my opinion it has to be done on this layer because we need connections between Entities in database.